src/elements/content-sidebar/AddTaskButton.js

```diff
@@ -45,7 +45,11 @@ class AddTaskButton extends React.Component<Props, State> {
         this.setState({ isTaskFormOpen: false, error: null });
     };
 
-    handleSubmitSuccess = () => this.setState({ isTaskFormOpen: false, error: null });
+    handleSubmitSuccess = () => {
```

Comment:
Can this be combined with, replaced by, or call the `handleModalClose` method?

Reply:
Hmm, so with this change `onTaskModalClose` is called for both Cancel and Submit (success), and it clears the autocomplete state stored up in `<ActivitySidebar>`. This handler is just closing the modal and resetting the error (and now calling to clear the autocomplete state).

I'm not sure of a way to combine since it needs to reset state at two different component levels. See https://github.com/box/box-ui-elements/blob/master/src/elements/content-sidebar/activity-feed/task-new/Task.js#L141-L148 for the other place that we're doing this same logic (for opening the same form for editing an existing task).